plugin/storage/es/spanstore/service_operation.go

```diff
@@ -96,29 +100,46 @@ func getServicesAggregation(maxDocCount int) elastic.Query {
 		Size(maxDocCount) // ES deprecated size omission for aggregating all. https://github.com/elastic/elasticsearch/issues/18838
 }
 
-func (s *ServiceOperationStorage) getOperations(ctx context.Context, indices []string, service string, maxDocCount int) ([]string, error) {
+func (s *ServiceOperationStorage) getOperations(ctx context.Context, indices []string, service, kind string, maxDocCount int) ([]spanstore.Operation, error) {
+	var searchService es.SearchService
+	if kind != "" {
+		serviceFilter := getOperationsAggregation(maxDocCount)
+		serviceNameQuery := elastic.NewTermQuery(serviceName, service)
+		serviceKindQuery := elastic.NewTermQuery(spanKind, kind)
+		serviceQuery := elastic.NewBoolQuery().Must(serviceNameQuery, serviceKindQuery)
+		searchService = s.client().Search(indices...).
+			Size(0).
+			Query(serviceQuery).
```

Comment:
the above decomposition is hard to follow, please inline it here with good formatting, e.g.
```suggestion
			Query(elastic.NewBoolQuery().Must(
				elastic.NewTermQuery(serviceName, service),
				elastic.NewTermQuery(spanKind, kind)
			)).
```